File: Editor3D/Python/in_memory_model/model_store.py
from ast import List


class ModelStore(IModelChanger):
    def __init__ (self, change_observers: List[IModelChangeObserver]):
        model_1 = PoligonalModel()
        scene_1 = Scene()
        flashe_1 = Flash()
        camera_1 = Camera()

        self.models = [model_1]
        self.scenes = [scene_1]
        self.flashes = [flashe_1]
        self.cameras = [camera_1]
        self.change_observers = change_observers

    def get_scene(self, id):
        for scene in scenes:
            if scene.id == id:
                return scene
        return None

    def notify_change(self, sender) -> None:
        return super().notify_change(sender)

File: Editor3D/Java/src/main/java/InMemoryModel/ModelStore.java
package InMemoryModel;

import ModelEliments.*;

import java.util.ArrayList;
import java.util.List;

public class ModelStore {
    public List<PoligonalModel> models;
    public List<Scene> scenes;
    public List<Flash> flashes;
    public List<Camera> cameras;
    private IModelChangeObserver[] changeObservers;

    public ModelStore(IModelChangeObserver[] changeObservers) throws Exception{
        this.changeObservers = changeObservers;

        this.models = new ArrayList<>();
        this.scenes = new ArrayList<>();
        this.flashes = new ArrayList<>();
        this.cameras = new ArrayList<>();

        List<Texture> textures = new ArrayList<>();

        this.models.add(new PoligonalModel(textures));
        this.scenes.add(new Scene(0, models, flashes, cameras));
        this.flashes.add(new Flash());
        this.cameras.add(new Camera());
    }

    public Scene getScene(int id)
    {
        return this.scenes.get(id);
    }

    public void notifyChange(IModelChanger sender) {
    }
}
